cpp/lib/src/Zotero.cc

```diff
@@ -703,6 +703,17 @@ void MarcFormatHandler::generateMarcRecord(MARC::Record * const record, const st
 
     if (not node_parameters.journal_name_.empty())
         record->insertField("JOU", { { 'a', node_parameters.journal_name_ } });
+
+    // remove any fields that match removal patterns
+    for (const auto &filter : site_params_->field_removal_filters_) {
+        auto field(filter.first);
+        if (record->fieldOrSubfieldMatched(filter.first, filter.second.get())) {
+            if (field.length() != 3)
+                field.erase(3);
+            record->erase(field);
```

Comment:
How about deleting the lines that are supposed to optionally delete the subfield code and rewriting this as "record->erase(tag_and_optional_subfield_code.substr(0, MARC::Record::TAG_LENGTH);"?